### PR TITLE
Added calculation of holdout data summary metrics for each horizon 

### PR DESCRIPTION
**Pull request by govarsha**
Added Calculation of Mean sMAPE, Median sMAPE, Mean MAPE, Median MAPE, Mean wMAPE, Median wMAPE values for each horizon in Holdout data summary metrics when horizon <= 10. 

image
image

### CODE REVIEW

**Review by ahosler — APPROVED** [review]
LGTM. I will open a second ticket to add some testing

**Review by codeloop — APPROVED** [review]

File: ads/opctl/operator/lowcode/forecast/utils.py [hunk 1]
```diff
@@ -44,6 +44,62 @@ def smape(actual, predicted) -> float:
     )
 
 
+def _build_metrics_per_horizon(data, outputs, target_columns, target_col):
```
Comment (mrDzurb) [inline ads/opctl/operator/lowcode/forecast/utils.py:47]: NIT: Could you add types for the function's attributes and return value type.

File: ads/opctl/operator/lowcode/forecast/utils.py [hunk 2]
```diff
@@ -44,6 +44,62 @@ def smape(actual, predicted) -> float:
...
+    for y_true, y_pred in zip(actuals_df.itertuples(index=False), forecasts_df.itertuples(index=False)):
+
+        y_true, y_pred = np.array(y_true), np.array(y_pred)
+
+        smapes = np.array([smape(actual=y_t, predicted=y_p) for y_t, y_p in zip(y_true, y_pred)])
+        mapes = np.array([mean_absolute_percentage_error(y_true=[y_t], y_pred=[y_p])  for y_t, y_p in zip(y_true, y_pred)])
+        wmapes = np.array([mape * weight for mape, weight in zip(mapes, wmape_weights)])
+        
+        metrics_row = {
+            "Mean sMAPE": np.mean(smapes),
```
Comment (mrDzurb) [inline ads/opctl/operator/lowcode/forecast/utils.py:88]: We use these keys in many places, I would suggest to move them into the const variables.

File: ads/opctl/operator/lowcode/forecast/utils.py [hunk 3]
```diff
@@ -44,6 +44,62 @@ def smape(actual, predicted) -> float:
...
+    totals = actuals_df.sum()
+    wmape_weights = np.array((totals / totals.sum()).values)
+
+    metrics_df = pd.DataFrame(columns=[
+        "Mean sMAPE", "Median sMAPE",
+        "Mean MAPE", "Median MAPE",
+        "Mean wMAPE", "Median wMAPE"
+    ])
+
+    for y_true, y_pred in zip(actuals_df.itertuples(index=False), forecasts_df.itertuples(index=False)):
```
Comment (mrDzurb) [inline ads/opctl/operator/lowcode/forecast/utils.py:79]: Nice :)

File: ads/opctl/operator/lowcode/forecast/utils.py (hunk 2 above)
Comment (govarsha) [reply ads/opctl/operator/lowcode/forecast/utils.py]: @mrDzurb I moved all metrics to const.py

File: ads/opctl/operator/lowcode/forecast/model/base_model.py [hunk 4]
```diff
@@ -295,24 +295,40 @@ def _test_evaluate_metrics(
...
-                "Median sMAPE": np.median(total_metrics.loc["sMAPE"]),
-                "Mean MAPE": np.mean(total_metrics.loc["MAPE"]),
-                "Median MAPE": np.median(total_metrics.loc["MAPE"]),
-                "Mean RMSE": np.mean(total_metrics.loc["RMSE"]),
-                "Median RMSE": np.median(total_metrics.loc["RMSE"]),
-                "Mean r2": np.mean(total_metrics.loc["r2"]),
-                "Median r2": np.median(total_metrics.loc["r2"]),
-                "Mean Explained Variance": np.mean(
-                    total_metrics.loc["Explained Variance"]
+                SupportedMetrics.MEAN_SMAPE: np.mean(total_metrics.loc[SupportedMetrics.SMAPE]),
```
Comment (mrDzurb) [inline ads/opctl/operator/lowcode/forecast/model/base_model.py:298]: Shouldn't be ```SupportedMetrics.SMAPE.value``` used here?

**Review by mrDzurb — APPROVED** [review]

File: ads/opctl/operator/lowcode/forecast/model/base_model.py (hunk 4 above)
Comment (ahosler) [reply ads/opctl/operator/lowcode/forecast/model/base_model.py:298]: I think what Varsha has should work because it's a class constant

**Review by ahosler — APPROVED** [review]

**Pull request merged**